cycle.go

```diff
@@ -55,6 +55,13 @@ func (e errCycleDetected) Error() string {
 	return b.String()
 }
 
+// IsCycleDetected returns a boolean as to whether the provided error indicates
+// a cycle was detected in the container graph.
+func IsCycleDetected(err error) bool {
+	_, ok := err.(errCycleDetected)
```

Comment:
You should probably be looking at the RootCause of the error. This won't return true for errors returned by invoke right now (because of the errWrapf).